src/pyodide-py/pyodide/_importhooks.py

```diff
@@ -0,0 +1,80 @@
+from ._core import JsProxy
+from importlib.machinery import ExtensionFileLoader
+from importlib.abc import MetaPathFinder
+from importlib.util import spec_from_loader
+import sys
+
+# From Python glossary: An importer is "both a finder and loader object."
+class JsImporter(MetaPathFinder, ExtensionFileLoader):
+    jsproxies = {}
+    ### Finder methods
+    @classmethod
+    def find_spec(cls, fullname, path, target=None):
+        [parent, _, child] = fullname.rpartition(".")
+        if parent:
+            parent_module = sys.modules[parent]
+            if not hasattr(parent_module, "__loader__") or not isinstance(
+                parent_module.__loader__, JsImporter
+            ):
+                # Not one of us.
+                return None
+            try:
+                jsproxy = getattr(parent_module, child)
+            except AttributeError:
+                raise ModuleNotFoundError(
+                    f"No module named {fullname!r}", name=fullname
+                ) from None
+            if not isinstance(jsproxy, JsProxy):
+                raise ModuleNotFoundError(
+                    f"No module named {fullname!r}", name=fullname
+                )
+        else:
+            try:
+                jsproxy = JsImporter.jsproxies[fullname]
+            except KeyError:
+                return None
+        loader = cls(fullname, jsproxy)
+        return spec_from_loader(fullname, loader, origin="javascript")
+
+    def __init__(self, name, jsproxy):
+        super().__init__(name, None)
+        self.jsproxy = jsproxy
+
+    def __repr__(self) -> str:
+        return "javascript module"
+
+    @classmethod
+    def find_module(cls, fullname, path):
+        raise NotImplementedError(
+            "find_module() is deprecated in favor of find_spec() since Python 3.4"
+        )
+
+    @staticmethod
+    def invalidate_caches() -> None:
+        pass
+
+    ### Loader methods
+    # Overwrite ExtensionFileLoader.create_module with our own mechanism
+    # that short circuits the file system access stuff.
+    # create_module_inner defined in jsimport.c
+    def create_module(self, spec):
+        return self.jsproxy
+
+    # use ExtensionFileLoader.exec_module (no override)
+
+    # used by importlib.util.spec_from_loader
+    def is_package(self, fullname):
+        return True  # ???
```

Comment:
What is the `???` issue here?

Reply:
I think it's right to say that these things are a `package`, I am just a little bit unclear on what a package is. I think it is just an issue of whether the syntax:
```python
from a.b import c`
```
is supported? If `a` is not a package, then this will raise an error? In that case I think `return True` is the correct behavior.

Reply:
That is my understanding too.